repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/glossary/CategoryPreProcessor.java

```diff
@@ -151,6 +162,18 @@ private void processUpdateCategory(AtlasEntity entity, AtlasVertex vertex) throw
 
         String newGlossaryQualifiedName = (String) anchor.getAttribute(QUALIFIED_NAME);
 
+        String lexicographicalSortOrder = (String) entity.getAttribute(LEXICOGRAPHICAL_SORT_ORDER);
+        String parentQname = "";
+        if(Objects.nonNull(parentCategory)) {
+            parentQname = (String) parentCategory.getAttribute(QUALIFIED_NAME);
+        }
+        if(StringUtils.isNotEmpty(lexicographicalSortOrder)) {
+            isValidLexoRank(entity.getTypeName(), lexicographicalSortOrder, newGlossaryQualifiedName, parentQname, this.discovery);
+        } else {
+            lexicographicalSortOrder = (String) storedCategory.getAttribute(LEXICOGRAPHICAL_SORT_ORDER);
+            entity.setAttribute(LEXICOGRAPHICAL_SORT_ORDER, lexicographicalSortOrder);
```

Comment:
I suggest pruning attribute here, remove it from entity
Also do the same for Term & Glossary as well

Reply:
used 
`entity.removeAttribute(LEXICOGRAPHICAL_SORT_ORDER);`
Hopefully this wont have any unwanted errors.

Reply:
I will not, but didn't you retest this scenario after the changes?